src/main/java/fiftyfive/administration/usermanagement/implemention/UserService.java

```diff
@@ -3,55 +3,80 @@
 import fiftyfive.administration.usermanagement.constant.Constant;
 import fiftyfive.administration.usermanagement.dto.CreateUserRequest;
 import fiftyfive.administration.usermanagement.dto.UpdateUserRequestData;
-import fiftyfive.administration.usermanagement.dto.User;
 import fiftyfive.administration.usermanagement.dto.UserResponseData;
+import fiftyfive.administration.usermanagement.entity.DeletedUser;
+import fiftyfive.administration.usermanagement.entity.User;
 import fiftyfive.administration.usermanagement.exception.RecordAlreadyExistsException;
 import fiftyfive.administration.usermanagement.exception.UserNotExistsException;
 import fiftyfive.administration.usermanagement.mapper.UserRequestMapper;
+import fiftyfive.administration.usermanagement.repository.DeletedUserRepository;
+import fiftyfive.administration.usermanagement.repository.UserRepository;
 import fiftyfive.administration.usermanagement.utility.UserValidation;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
-
 import java.util.List;
+import java.util.Optional;
 
 @Service
 public class UserService {
 
     @Autowired
-    final UserRequestMapper userRequestMapper;
+    UserRequestMapper userRequestMapper;
 
     @Autowired
     UserValidation userValidation;
 
-    public UserService(UserRequestMapper userRequestMapper) {
-        this.userRequestMapper = userRequestMapper;
-    }
+    @Autowired
+    UserRepository userRepository;
+
+    @Autowired
+    DeletedUserRepository deletedUserRepository;
+
 
-    public UserResponseData createUser(CreateUserRequest createUserRequest, List<User> users) throws RecordAlreadyExistsException {
-        for (User existingUser : users) {
-            if (existingUser.getUsername().equals(createUserRequest.getUsername())) {
-                throw new RecordAlreadyExistsException(String.format(Constant.USER_ALREADY_EXISTS, createUserRequest.getUsername()));
-            }
+    public UserResponseData createUser(CreateUserRequest createUserRequest) throws RecordAlreadyExistsException {
+        String username = createUserRequest.getUsername();
+        User user = userRepository.findByUsername(username);
+        if (user != null) {
+            throw new RecordAlreadyExistsException(String.format(Constant.USER_ALREADY_EXISTS, username));
         }
-        return userRequestMapper.createUserMapper(createUserRequest, users);
+        user = userRequestMapper.mapToUser(createUserRequest);
+        userRepository.save(user);
+        createUserRequest.setId(user.getId());
+        return userRequestMapper.createUserResponseMapper(createUserRequest);
```

Comment:
Instead of having a request to response mapping. We can have a model to response mapping.